internal/git/fetch_tags.go

```diff
@@ -0,0 +1,44 @@
+package git
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/charmbracelet/log"
+	"github.com/sourcegraph/scip-go/internal/command"
+)
+
+// FetchTagsIfMissing attempts to fetch all tags for a git repository,
+// since we rely on git tags (similar to other Go tooling) to determine
+// the module version number (see infer_module_version.go).
+//
+// Since this can change on-disk repository state, this operation should
+// only be invoked during auto-indexing.
+func FetchTagsIfMissing(dir string) error {
+	tagsDir := path.Join(dir, ".git", "refs", "tags")
+	stat, err := os.Stat(tagsDir)
+	dirExists := err == nil && stat.IsDir()
+	if dirExists {
+		tags, err := os.ReadDir(tagsDir)
+		if err != nil {
+			return err
+		}
+		if len(tags) > 0 {
+			return nil // tags already exist, don't fetch more
+		}
+	}
+	// In an auto-indexing context, the remote is called origin.
+	// See https://sourcegraph.sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/cmd/executor/internal/worker/workspace/clone.go?L31:1-31:15
+	out, err := command.Run(dir, "git", "fetch", "origin", "refs/tags/*:refs/tags/*")
+	if strings.Count(out, "new tag") == 0 {
```

Comment:
I see there is some affordance in the executor code for allow it to fetch tags by passing the `FetchTags` option on the [Job](https://sourcegraph.sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/cmd/executor/internal/worker/workspace/clone.go?L89).  


Would it be possible to conditionally have that property set for a scip-go indexer job?

Reply:
To get that to work, we have two options:

- Do a database migration (by adding that as a field for `AutoIndexJob`) + modify the auto-indexing Lua script a bit + modify the Lua bridging code a bit. <- This would allow any indexer to request tags.
- Modify the existing hack for lsif-go to. <- I do not want to do this.

Doing a database migration for such a situation is overkill. In other auto-indexing scripts, we do things like dependency fetching. In a way, the tags can be thought of as a data dependency of the indexer. So it's not THAT weird for us to do that here.

Reply:
Another alternate solution, which is probably the smallest patch, is to modify `go.lua` directly to add a tag fetching step.